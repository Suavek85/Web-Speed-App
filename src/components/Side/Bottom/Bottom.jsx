import React from 'react'
import { useSelector } from "react-redux"
import SideCard from '../SideCard'
import { RESULT } from '../../../constants/content.js'

export default function Bottom() {

  const getDataState = useSelector(state => state.stateReducer.getData)

  const content = {
    header: 'Unused CSS',
    mainContent: 'Remove dead rules from stylesheets and defer the loading of not used CSS.',
    result: [RESULT],
  }

  const data = [getDataState[6]]
  const position = 'middle'
  const props = { content, data, position }

  return (
    <SideCard {...props} />
  )
}

