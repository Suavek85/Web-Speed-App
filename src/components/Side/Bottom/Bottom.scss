@import '../../../scss/styles.scss';

.resultsbottom__wrapper {
    width: 100%;
    height: 50%;
    display: flex;
    flex-direction: column;
    background-color: $color-bottom-background;
}

.resultsbottom__content-wrapper {
    padding: 0px 30px;
    > h1 {
      color: lightgrey;
    }
    > p:nth-child(2) {
      color: lightgrey;
    }
    > p:nth-child(3) {
      color: ivory;
      span {
        color: ivory;
      }
    }
}

:export {
    blockWrapper: resultsbottom__wrapper;
    blockRow: resultsbottom__row;
    contentWrapper: resultsbottom__content-wrapper;
}