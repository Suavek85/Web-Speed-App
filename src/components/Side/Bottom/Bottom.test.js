import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import Bottom from './Bottom'

describe('Bottom snapshot', () => {

  it('should match the snapshot', () => {
    const mockedResponse = [0.67, 0.47, "https://www.bbc.com/", "SLOW", "AVERAGE", "FAST", 0.98]
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue(mockedResponse)
    const wrapper = shallow(<Bottom />)
    expect(wrapper).toMatchSnapshot()
  })

})