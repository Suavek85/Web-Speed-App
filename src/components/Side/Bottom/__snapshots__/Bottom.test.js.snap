// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom snapshot should match the snapshot 1`] = `
<Section
  content={
    Object {
      "header": "Unused CSS",
      "mainContent": "Remove dead rules from stylesheets and defer the loading of not used CSS.",
      "result": Array [
        "Result: ",
      ],
    }
  }
  data={
    Array [
      0.98,
    ]
  }
  position="middle"
/>
`;
